CMakeLists.txt

```diff
@@ -256,6 +258,39 @@ if (COG_PLATFORM_FDO AND NOT COG_USE_WEBKITGTK)
         add_definitions(-DCOG_IM_API_SUPPORTED=0)
     endif ()
 
+    if (COG_WESTON_DIRECT_DISPLAY)
+        file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/weston)
+        pkg_check_modules(LIBWESTON REQUIRED libweston-9-protocols)
+        pkg_get_variable(LIBWESTON_PKG_DATA_DIR libweston-9-protocols "pkgdatadir")
+        find_package(WaylandScanner REQUIRED)
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/weston/weston-direct-display-client-protocol.h
+                   ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
+            DEPENDS ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml
+            COMMAND ${WAYLAND_SCANNER} client-header < ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml > ${CMAKE_BINARY_DIR}/weston/weston-direct-display-client-protocol.h
+            COMMAND ${WAYLAND_SCANNER} ${WAYLAND_SCANNER_CODE_ARG} < ${LIBWESTON_PKG_DATA_DIR}/weston-direct-display.xml > ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
+            VERBATIM
+        )
+
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/weston/weston-content-protection-client-protocol.h
+                   ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
+            DEPENDS ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml
+            COMMAND ${WAYLAND_SCANNER} client-header < ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml > ${CMAKE_BINARY_DIR}/weston/weston-content-protection-client-protocol.h
+            COMMAND ${WAYLAND_SCANNER} ${WAYLAND_SCANNER_CODE_ARG} < ${LIBWESTON_PKG_DATA_DIR}/weston-content-protection.xml > ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
+            VERBATIM
+        )
+
+        list(APPEND COGPLATFORM_FDO_SOURCES
+            ${CMAKE_BINARY_DIR}/weston/weston-direct-display-protocol.c
+            ${CMAKE_BINARY_DIR}/weston/weston-content-protection-protocol.c
+        )
+        list(APPEND COGPLATFORM_FDO_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/weston)
+        add_definitions(-DCOG_ENABLE_WESTON_DIRECT_DISPLAY=1)
```

Comment:
That doesn't seem doable because the target is defined afterwards. I can't define the target before either because when direct display is enabled I need to add compilation units in the target.

Reply:
Yeah, actually I've found a way to do this using an intermediary variable.

Reply:
I think it should be even possible to move creation of the target to be done earlier. All the `target_<foo>()` CMake functions are cumulative, and the order in which they are called on the target is irrelevant.